### PR TITLE
C110 added a new end point to filter and fetch the tweets

### PR DESCRIPTION
**Pull request by SupriyaPamulapati**

### CODE REVIEW

File: src/main/java/UTwitter/service/TwitterImplement.java [hunk 1]
```diff
@@ -50,26 +52,46 @@ public Status sendTweet(String args) throws TwitterException {
...
+                Format format = new SimpleDateFormat("dd-mm-yyy HH:mm:ss");
+                String date = format.format(createdAt);
+                String twitterHandle = status.getUser().getScreenName();
+                twitterResponseModel = new TwitterResponseModel(message, name, twitterHandle, profileImageUrl, date);
+                arrayList.add(twitterResponseModel);
             }
         } catch (TwitterException e) {
             log.error("error in retrieving tweets ");
         }
         return arrayList;
```
Comment (anasuya1117) [inline src/main/java/UTwitter/service/TwitterImplement.java:69]: arrayList is a data type in java, it's not a good variable name.

File: src/main/java/UTwitter/service/TwitterImplement.java [hunk 2]
```diff
@@ -50,26 +52,46 @@ public Status sendTweet(String args) throws TwitterException {
...
+                Date createdAt = status.getCreatedAt();
+                Format format = new SimpleDateFormat("dd-mm-yyy HH:mm:ss");
+                String date = format.format(createdAt);
+                String twitterHandle = status.getUser().getScreenName();
+                twitterResponseModel = new TwitterResponseModel(message, name, twitterHandle, profileImageUrl, date);
+                arrayList.add(twitterResponseModel);
+            }
+        } catch (TwitterException e) {
+            log.error("error in retrieving tweets ");
+        }
```
Comment (anasuya1117) [inline src/main/java/UTwitter/service/TwitterImplement.java:91]: Apply DRY principle

File: src/test/com/test/GetFilteredTweets_Test.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        str.add("good start if day");
+        str.add("A. good day");
+        str.add("good and easy life");
+        str.add("good to b good");
+        when(tweetPost.filteredTweets("good")).thenReturn(Response.ok(str).build());
+        String[] s = new String[]{"good start if day", "A. good day", "good and easy life", "good to b good"};
+        Response expectedTweet = Response.ok(s).build();
+        Response actualTweet = tweetPost.filteredTweets("good");
+        Assert.assertEquals(expectedTweet.getStatus(), actualTweet.getStatus());
+        Assert.assertEquals(expectedTweet.getStatus(), actualTweet.getStatus());
```
Comment (anasuya1117) [inline src/test/com/test/GetFilteredTweets_Test.java:59]: match the data, status will be 200 ok for all cases.

File: src/test/com/test/GetFilteredTweets_Test.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Test
+    public void noTweetMatch_Test() throws TwitterException {
+        ResponseList<Status> responseList = mock(ResponseList.class);
+        when(responseList.size()).thenReturn(0);
+        when(twitter.getHomeTimeline()).thenReturn(responseList);
+        List<TwitterResponseModel> actual = twitterImplement.getFilteredTweets("forest");
+        Assert.assertEquals(Arrays.asList(), actual);
```
Comment (anasuya1117) [inline src/test/com/test/GetFilteredTweets_Test.java:68]: Test is not clear. please update

**Review by anasuya1117 — COMMENTED** [review]
Update the test cases and address the comments

File: src/main/java/UTwitter/resources/Controller.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package UTwitter.resources;
 
 import UTwitter.service.TwitterImplement;
```
Comment (PoojaMohite-Khoros) [inline src/main/java/UTwitter/resources/Controller.java:3]: UTwitter , if this is package name , its not according to naming convention, please correct

File: src/test/com/test/GetFilteredTweets_Test.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetFilteredTweets_Test {
```
Comment (PoojaMohite-Khoros) [inline src/test/com/test/GetFilteredTweets_Test.java:27]: File names usually doesn't have underscores so please check with 
Naming conversion link that I had sent in channel

**Review by PoojaMohite-Khoros — CHANGES_REQUESTED** [review]

**Review by PoojaMohite-Khoros — APPROVED** [review]
We can always improve the filter.
Please merge and create PR for next lab

**Pull request merged**